### ISSUE TITLE
Refactor Bayes Factor computation out of az.plot_bf

### ISSUE THREAD

**Issue by mpmbq2** [arviz-devs/arviz#2402]
The function `az.plot_bf` is very useful for visualizing Bayes Factors. However, some users may want to use Bayes Factors outside the context of the plot, so it would be useful to refactor the logic to compute the BF from `arviz.plots` to `arviz.stats` and make it part of the public API. 

This seems like a very small change, and I'm happy to do it myself. But I am new to open source and have not communicated with you all before. So before I make the PR I wanted to make sure there's not a good reason to keep it the way it is right now.

**Comment by aloctavodia** [comment]
Hi @mpmbq2, thanks for offering help. Your suggestion is a good one and you can send the PR if you want.

I just want to let you know that we are refactoring ArviZ. As part of the refactor, we are splitting functionality into 3 sub-packages and adding most of the new features in those packages (although there is no strict rule against adding new features to the main repo). 

In the context of your proposal, the computation of the BF  should live in [arviz-stats](https://github.com/arviz-devs/arviz-stats) and the plotting should be on [arvivz-plots](https://github.com/arviz-devs/arviz-plots). Making a PR into one or both of those packages will be more work than making the changes here.

Both options will be helpful for us and the community, so please select the one that you like the most.

**Comment by mpmbq2** [comment]
Thanks for the heads up RE the package migration. For the sake of not over-thinking it, I decided to go ahead and make a PR to this repo. Once I get it accepted, I'll have a look at submitting PRs to the stats and plots repos.

**Comment by PiyushPanwarFST** [comment]
Sir, I would like to work on this issue. Could you please let me know if anyone else is already working on it?

**Comment by aloctavodia** [comment]
Hi @PiyushPanwarFST thanks for your help. There is a PR already open addressing this issue for this repository. Maybe you work on addressing this issue on [arviz-stats](https://github.com/arviz-devs/arviz-stats) instead.

**Comment by PiyushPanwarFST** [comment]
Sir can you please give me the clarity where I should contribute regarding the Bayes Factor functionality. From what I understand:

The work on the main repository (arviz) is already completed.
I should focus on arviz-stats for this functionality.
I also noticed there isn’t an open issue related to Bayes Factor computation in arviz-stats. Should I create an issue for it, or can I directly raise a PR to add the functionality? Also, should I clone the arviz-stats repository and start contributing there directly?

Looking forward to your guidance!

**Comment by aloctavodia** [comment]
Yes, this is being addressed in the main repo, then better to focus on arviz-stats. You can open an issue if you want. Yes clone the repo and contribute there.
